tests/unit/test_sns.py

```diff
@@ -609,13 +609,33 @@ def test_filter_policy_on_message_body(self):
                     ({"f1": ["v3", "v4"], "f2": "v5"}, False),
                 ),
             ),
+            (
+                {"f1": {"f2": {"f3": {"f4": ["v1"]}}}},
+                (
+                    ({"f1": {"f2": {"f3": {"f4": "v1"}}}}, True),
+                    ({"f1": [{"f2": {"f3": {"f4": "v1"}}}]}, True),
+                    ({"f1": [{"f2": [{"f3": {"f4": "v1"}}]}]}, True),
+                    ({"f1": [{"f2": [[{"f3": {"f4": "v1"}}]]}]}, True),
+                    ({"f1": [{"f2": [{"f3": {"f4": "v1"}, "f5": {"f6": "v2"}}]}]}, True),
+                    ({"f1": [{"f2": [[{"f3": {"f4": "v2"}}, {"f3": {"f4": "v1"}}]]}]}, True),
+                    ({"f1": [{"f2": {"f3": {"f4": "v2"}}}]}, False),
+                    ({"f1": [{"f2": {"fx": {"f4": "v1"}}}]}, False),
+                    ({"f1": [{"fx": {"f3": {"f4": "v1"}}}]}, False),
+                    ({"fx": [{"f2": {"f3": {"f4": "v1"}}}]}, False),
+                    ({"f1": [{"f2": [{"f3": {"f4": "v2"}, "f5": {"f6": "v3"}}]}]}, False),
+                    ({"f1": [{"f2": [[{"f3": {"f4": "v2"}}, {"f3": {"f4": "v3"}}]]}]}, False),
```

Comment:
Good question 🤔 
I think the followings, I need to check the full cases, I hope I'll somewhat soon have time to implement https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-sns-message-additional-filtering-operators/

```
{"f1": [{"f2": [[{"f3": {"f4": "v1"}}]]}]}
{"f1": [{"f2": [{"f3": {"f4": "v1"}, "f5": {"f6": "v2"}}]}]}
# this one has a list of a list and it still worked!
{"f1": [{"f2": [[{"f3": {"f4": "v2"}}, {"f3": {"f4": "v1"}}]]}]}
```

When I'll be implementing the new features for SNS, I can look into testing them with the test event pattern too, and adding a few from SNS if they are missing 👍

Reply:
wow, the `list of a list` is interesting 😮 . It would be great if you can add them to the event pattern test suite 👍
It's as easy as copy/pasting one of the json5 templates 😎 (just need to keep some of the mandatory fields for EventBridge for API compatibility; see minimal.json5)
Thank you for thinking about all these edge cases 💯